### PR TITLE
support updating Tags on Role resources

### PR DESCRIPTION
**Pull request by jaypipes**
Adds support for updating (adding/removing) Tags from a Role resource in
the IAM controller.

Unfortunately, the "standard" Tagris
ListTagsForResource/TagResource/UntagResource AWS Resource Group Tagging
API calls aren't supported for IAM. You need to use the IAM-specific
TagRole/UntagRole/ListRoleTags IAM API calls to work with tags on Role
resources.

Following patches will add support for the same tags on Policy
resources.

Issue: aws-controllers-k8s/community#1119

Signed-off-by: Jay Pipes <jaypipes@gmail.com>

By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/role/hooks.go [hunk 1]
```diff
@@ -132,3 +134,144 @@ func (rm *resourceManager) detachPolicy(
...
+		resp, err = rm.sdkapi.ListRoleTagsWithContext(ctx, input)
+		if err != nil || resp == nil {
+			break
+		}
+		for _, t := range resp.Tags {
+			res = append(res, &svcapitypes.Tag{Key: t.Key, Value: t.Value})
+		}
+		if resp.IsTruncated != nil && !*resp.IsTruncated {
+			break
+		}
```
Comment (RedbackThomson) [inline pkg/resource/role/hooks.go:227]: If you are attempting to use pagination for `ListRoleTagsWithContext`, then I think you are missing the `input.Marker` field from this `resp` object.

**Review by RedbackThomson — CHANGES_REQUESTED** [review]
Implementation looks good. Just need to fix that pagination

**Review by vijtrip2 — APPROVED** [review]
/lgtm

File: pkg/resource/role/hooks.go (hunk 1 above)
Comment (jaypipes) [reply pkg/resource/role/hooks.go:227]: Hmm, yeah. Luckily the limit on tags by default is 50 so in practical matters this won't occur. But, I will fix in a followup PR, thanks for the eagle eyes!

**Pull request merged**